config/entry.go

```diff
@@ -34,19 +34,19 @@ func makeEntryFromValue(value any) *Entry {
 }
 
 func (e *Entry) validate(key string) error {
-	if e.Value == nil {
-		if e.Required {
-			return errors.Errorf("%q is required and must contain a value", key)
-		}
-		// When just nil, we don't need to validate.
-		return nil
-	}
-
 	if err := e.tryEnvVarConversion(key); err != nil {
 		return err
 	}
 
+	v := reflect.ValueOf(e.Value)
+	if e.Required && (!v.IsValid() || e.Value == nil) {
+		return errors.Errorf("%q is required", key)
+	}
+
 	t := reflect.TypeOf(e.Value)
+	if t == nil {
+		return nil // Can't determine
+	}
```

Comment:
```suggestion
	t := v.Type()
```

Reply:
This won't work because

```
panic: reflect: call of reflect.Value.Type on zero Value
```

This happens when asserting the config `server.proxy.host`. Which is `nil` and also not required. Imo it's fine if we don't continue with such a value.

Reply:
It makes sense indeed. No need to validate in that case.